pkg/metrics/util.go

```diff
@@ -22,3 +23,57 @@ func EventPayload(ev types.Event) ([]byte, error) {
 	payload := map[string]interface{}{"event": ev, "versions": vmap}
 	return json.Marshal(payload)
 }
+
+var secretKeywords = []string{"password", "secret", "token", "key"}
+
+// redactFlags redacts presumed secret values from a string slice based on preset keywords.
+// For example, given ["--license", "./path", "--password", "123", "--a-secret=123"]
+// it will return ["--license", "./path", "--password", "*****", "--a-secret=*****"]"
+func redactFlags(flags []string) []string {
+	result := make([]string, len(flags))
+
+	valueHasSecret := false
+	for i, flag := range flags {
+		// The previous iteration detected one of the secret keywords on flag key, let's redact this value
+		if valueHasSecret {
+			result[i] = "*****"
+			valueHasSecret = false
+			continue
+		}
+		// Check if this is the flag name or the value
+		isFlagKey := strings.HasPrefix(flag, "-")
+		// Flag is of the form --key=value
+		if isFlagKey && strings.Contains(flag, "=") {
```

Comment:
I was hopping to avoid this "complex" parsing logic by reusing something but it so happens that most of what I could find is built around parsing flags for a set of predefined keys, not an arbitrary set like we're trying to do here. (i.e. the standard flag package - https://pkg.go.dev/flag)